src/settings.ts

```diff
@@ -0,0 +1,52 @@
+import { WorkspaceConfiguration, workspace, window, commands } from "vscode";
+
+let oldConfig: WorkspaceConfiguration = getXMLConfiguration();
+let javaHomeConfig = getJavaHomeConfiguration();
+
+export function getXMLConfiguration(): WorkspaceConfiguration {
+  return workspace.getConfiguration("xml");
+}
+
+export function getJavaHomeConfiguration() {
+  return workspace.getConfiguration("java.home");
+}
+
+export function onConfigurationChange() {
+  return workspace.onDidChangeConfiguration(params => {
+
+    if(!params.affectsConfiguration("xml") && !params.affectsConfiguration("java.home")) {
+      return;
+    }
+
+    const newConfig = getXMLConfiguration();
+
+    if(hasPreferenceChanged(oldConfig, newConfig, "java.home")) { // checks "xml.java.home"
+      const message = "`xml.java.home` path has changed. Please restart VS Code."
+      createReloadWindowMessage(message);
+    }
+
+    else if(javaHomeConfig != getJavaHomeConfiguration()) { // checks "java.home"
```

Comment:
on check it if xml.java.home is not set, else you can ignore the change, it doesn't impact XML